@types/discordBot.d.ts

```diff
@@ -0,0 +1,70 @@
+import { BotErrorType, IdType } from '../helpers/discordBot'
+
+export type snowflake = string
+
+interface BotErrorBase {
+  type: BotErrorType
+  message: string
+}
+
+export interface BotAuthError extends BotErrorBase {
+  type: BotErrorType.auth_error
+}
+
+export interface BotValidationError extends BotErrorBase {
+  type: BotErrorType.validation_error
+  issues: { [key: string]: string }[]
+}
+
+export interface BotApiError extends BotErrorBase {
+  type: BotErrorType.api_error
+}
+
+export type BotError = BotAuthError | BotValidationError | BotApiError
+
+export type MessageResponse =
+  | {
+      id: snowflake
+    }
+  | BotError
+
+export interface MessageEmbed {
+  title?: string
+  description?: string
+  url?: string
+  timestamp?: number
+  color?: any
+  fields?: any[]
+  files?: any[]
+  author?: any
+  thumbnail?: any
+  image?: any
+  video?: any
+  footer?: any
```

Comment:
https://github.com/discordjs/discord.js/blob/master/typings/index.d.ts#L3546-L3558 it depends on many other types I'm not sure how deep to go

Reply:
could you add a comment with the link? 

couldn't we import discord.js's types instead of recreating this? Of no, then perhaps instead of `any`, we just type the specific resource that we need. If in the future someone needs more, they can add it in by following your comment

Reply:
As discussed, `discord.js` has been added as a dependency and relevant types are now imported from it rather than been manually recreated.